src/RealtimeClient.ts

```diff
@@ -452,11 +467,28 @@ export default class RealtimeClient {
     this.log('transport', `connected to ${this._endPointURL()}`)
     this._flushSendBuffer()
     this.reconnectTimer.reset()
-    this.heartbeatTimer && clearInterval(this.heartbeatTimer)
-    this.heartbeatTimer = setInterval(
-      () => this._sendHeartbeat(),
-      this.heartbeatIntervalMs
-    )
+    if (!this.worker) {
+      this.heartbeatTimer && clearInterval(this.heartbeatTimer)
+      this.heartbeatTimer = setInterval(
+        () => this._sendHeartbeat(),
+        this.heartbeatIntervalMs
+      )
+    } else {
+      this.log('worker', `starting worker for from ${this.workerUrl!}`)
+
+      this.workerRef = new Worker(this.workerUrl!)
+      this.workerRef.onerror = (error) => {
+        this.log('worker', 'worker error', error.message)
+        this.workerRef!.terminate()
+      }
+      this.workerRef.onmessage = (event) => {
+        if (event.data === 'keepAlive') {
+          this._sendHeartbeat()
+        }
+      }
+      this.workerRef.postMessage('start')
```

Comment:
instead of hardcoding the setInterval time to `30000` in worker.js can we pass it in so it takes in `heartbeatIntervalMs` from the client?

so instead of start we pass in this.heartbeatIntervalMs.